torch/csrc/distributed/c10d/ProcessGroupNCCL.hpp

```diff
@@ -89,13 +89,22 @@ static std::vector<std::string> TORCH_NCCL_WAIT_TIMEOUT_DUMP_MILSEC = {
 static std::vector<std::string> TORCH_NCCL_COORD_CHECK_MILSEC = {
     "TORCH_NCCL_COORD_CHECK_MILSEC"};
 
+// Whether to abort the communicators when users call destroy_process_group().
+// If yes, communicators will be aborted when destroy_process_group is called,
+// but not in destructor.
+static std::vector<std::string> TORCH_NCCL_ABORT_IN_DESTROY_PG = {
+    "TORCH_NCCL_ABORT_IN_DESTROY_PG"};
```

Comment:
In ideal case, calling `ncclCommDestroy` (which is collective) is better than calling `ncclCommAbort` in normal termination. We can make the change in separate PR, but maybe better to avoid using "ABORT" in user facing control variable?

Reply:
what would you propose for the env name?

Reply:
Note that 'abort' is not naming after NCCL level ncclCommAbort, but PG level abort(), we can switch to ncclCommDestroy under the hood without changing user interfaces. 

Also note this ENV is here just for the roll out purpose, the plan is to delete it after the rollout